File: bot/commands/util/stats.js
const ms = require('parse-ms')
exports.run = async (bot) => {
  bot.registerCommand('stats', async (message, args) => {
    await bot.checkUserAndGuild(message)
    if (bot.checkBannedUsers(message.author.id)) return

    let userStats
    if (!args[0]) {
      userStats = message.author
    } else {
      userStats = await bot.getUser(message, args.join(' '))
    }

    bot.database.Userdata.findOne({ userID: userStats.id }, async (err, userdata) => {
      if (err) bot.log('error', err)

      if (userdata && userdata.stats) {
        const hbTime = ms(userdata.stats.totalHuntbotTime)
        const statsEmbed = {
          embed: {
            author: {
              name: `${userStats.username}'s stats`,
              icon_url: userStats.avatarURL
            },
            color: bot.getEmbedColor(bot, message),
            description: '*Since 10 April 2020 at 7:34 PM EDT*',
            fields: [
              {
                name: 'OwO Count',
                value: `\`${userdata.stats.owoCount}\``,
                inline: true
              },
              {
                name: 'Hunt Count',
                value: `\`${userdata.stats.huntCount}\``,
                inline: true
              },
              {
                name: 'Battle Count',
                value: `\`${userdata.stats.battleCount}\``,
                inline: true
              },
              {
                name: 'Pray/Curse Count',
                value: `\`${userdata.stats.praycurseCount}\``,
                inline: true
              },
              {
                name: 'Completed Huntbots',
                value: `\`${userdata.stats.completedHuntbots}\``,
                inline: true
              },
              {
                name: 'Total Huntbot Time',
                value: `\`${hbTime.days}D ${hbTime.hours}H ${hbTime.minutes}m\``,
                inline: true
              },
              {
                name: '\u200B',
                value: '\u200B',
                inline: false
              },
              {
                name: 'Daily OwO Count',
                value: `\`${userdata.stats.dailyOwoCount}\``,
                inline: true
              },
              {
                name: 'Daily Hunt Count',
                value: `\`${userdata.stats.dailyHuntCount}\``,
                inline: true
              },
              {
                name: 'Daily Battle Count',
                value: `\`${userdata.stats.dailyBattleCount}\``,
                inline: true
              },
              {
                name: 'Daily Pray/Curse Count',
                value: `\`${userdata.stats.dailyPraycurseCount}\``,
                inline: true
              }
            ],
            timestamp: new Date()
          }
        }
        bot.createMessage(message.channel.id, statsEmbed)
      }
    })
  }, {
    aliases: ['s'],
    cooldown: 3000,
    cooldownMessage: 'Whoa there slow down, the cooldown is 3 seconds!'
  })
}

File: src/main/java/dastrangeboi/atlantis/Commands/Stats.java
package dastrangeboi.atlantis.Commands;

import dastrangeboi.atlantis.Database;
import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent;
import org.bson.Document;

public class Stats {

    private static String getStats(Document userdata, String statName) {
        Document stats = (Document) userdata.get("stats");
        return "`"+stats.getInteger(statName)+"`";
    }

    public static void run(GuildMessageReceivedEvent message) {
        Document userdata = Database.getUserdata(message.getAuthor().getId());
        EmbedBuilder statsEmbed = new EmbedBuilder()
                .setAuthor(message.getAuthor().getName()+"'s stats", null, message.getAuthor().getAvatarUrl())
                .setColor(message.getGuild().getSelfMember().getColor())
                .addField("OwO Count", getStats(userdata, "owoCount"), true)
                .addField("Hunt Count", getStats(userdata, "huntCount"), true)
                .addField("Battle Count", getStats(userdata, "battleCount"), true)
                .addField("Pray/Curse Count", getStats(userdata, "praycurseCount"), true)
                .addField("Completed Huntbots", getStats(userdata, "completedHuntbots"), true)
                .addField("Total Huntbot Time", getStats(userdata, "totalHuntbotTime"), true)
                .addBlankField(false)
                .addField("Daily OwO Count", getStats(userdata, "dailyOwoCount"), true)
                .addField("Daily Hunt Count", getStats(userdata, "dailyHuntCount"), true)
                .addField("Daily Battle Count", getStats(userdata, "dailyBattleCount"), true)
                .addField("Daily Pray/Curse Count", getStats(userdata, "dailyPraycurseCount"), true);

        message.getChannel().sendMessage(statsEmbed.build()).queue();
    }
}
